rosette/src/BaseSupp.cc

```diff
@@ -742,24 +744,29 @@ DEF("regexpCompare", regexpCompare, 2, 2) {
     CHECK(0, RBLstring, re);
     CHECK(1, RBLstring, str);
 
-    char *re_s, *str_s;
-    regexp* r;
     Ob* rv;
 
-    re_s = (char*)&re->byte(0);
-    str_s = (char*)&str->byte(0);
+    char* re_s = (char*)&re->byte(0);
+    char* str_s = (char*)&str->byte(0);
+
+    try {
+        std::regex r(re_s); // Compile the regexp
 
-    if ((r = GS_regcomp(re_s)) && GS_regexec(r, str_s)) {
-        int len = r->endp[0] - r->startp[0];
-        char* res = (char*)malloc(len + 1);
-        strncpy(res, r->startp[0], len);
-        res[len] = '\0';
-        rv = RBLstring::create(res);
-        free(res);
+        if (std::regex_match(str_s, r) == true) {   // See if it matches
```

Comment:
You don't need the `== true`.

Reply:
I'm just trying to make things clear.  I'll take it out.

Done